gandi/resource_nameservers.go

```diff
@@ -0,0 +1,119 @@
+package gandi
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/go-gandi/go-gandi/domain"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceNameservers() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNameserversCreate,
+		Read:          resourceNameserversRead,
+		UpdateContext: resourceNameserversUpdate,
+		Delete:        resourceNameserversDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The FQDN of the domain",
+			},
+			"nameservers": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "A list of nameservers for the domain",
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{Default: schema.DefaultTimeout(1 * time.Minute)},
+	}
+}
+
+func resourceNameserversCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients).Domain
+
+	domain := d.Get("domain").(string)
+	d.SetId(domain)
+	nameservers := expandNameServers(d.Get("nameservers").([]interface{}))
+
+	if err := client.UpdateNameServers(domain, nameservers); err != nil {
+		return diag.FromErr(err)
+	}
+
+	err := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		return retryableGetNameServers(client, domain, nameservers)
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return diag.FromErr(resourceNameserversRead(d, meta))
+}
+
+func resourceNameserversRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients).Domain
+	domain := d.Id()
+	nameservers, err := client.GetNameServers(domain)
+	if err != nil {
+		d.SetId("")
+		return err
+	}
+	if err = d.Set("nameservers", nameservers); err != nil {
+		return fmt.Errorf("failed to set nameservers for %s: %w", d.Id(), err)
+	}
```

Comment:
I suggest you add:

```suggestion
	}
	if err = d.Set("domain", domain); err != nil {
	    return fmt.Errorf("failed to set domain name for %s: %w", d.Id(), err)
	}
```

Otherwise, if you import then plan, you still have a change:

```
Terraform will perform the following actions:

  # gandi_nameservers.enxio_fr will be updated in-place
  ~ resource "gandi_nameservers" "enxio_fr" {
      + domain      = "enxio.fr"
        id          = "enxio.fr"
        # (1 unchanged attribute hidden)

        # (1 unchanged block hidden)
    }
```

It will make Pulumi happy too. ;-)

Reply:
Hehe, that's actually now looks evident!

It's applied and thanks for the fix!